src/docfx/lib/ProcessUtility.cs

```diff
@@ -45,32 +45,36 @@ public static async Task<bool> IsExclusiveLockHeld(string lockName)
         /// Acquire a shared lock for input lock name
         /// The returned `acquirer` are used for tracking the acquired lock, instead of thread info, since the thread info may change in asynchronous programming model
         /// </summary>
-        public static async Task<(bool acquired, string acquirer)> AcquireSharedLock(string lockName)
+        public static Task<(bool acquired, string acquirer)> AcquireSharedLock(string lockName, bool retryUntilSuccess = false)
         {
             Debug.Assert(!string.IsNullOrEmpty(lockName));
-
-            var acquired = false;
-            var acquirer = (string)null;
             var lockPath = GetLockFilePath(lockName);
 
-            await ReadAndWriteFile<LockInfo>(lockPath, lockInfo =>
-            {
-                lockInfo = lockInfo ?? new LockInfo();
-                lockInfo = FilterExpiredAcquirers(lockInfo);
+            return retryUntilSuccess ? AcquireWithRetry(lockName, AcquireLock) : AcquireLock();
 
-                if (lockInfo.Type == LockType.Exclusive)
+            async Task<(bool, string)> AcquireLock()
+            {
+                var acquired = false;
+                var acquirer = (string)null;
+                await ReadAndWriteFile<LockInfo>(lockPath, lockInfo =>
                 {
-                    return Task.FromResult(lockInfo);
-                }
+                    lockInfo = lockInfo ?? new LockInfo();
```

Comment:
put this line into `FilterExpiredAcquirers`? then we can get rid of the null assertion

Reply:
filter  is to filter existing timeout lock, not to create a new lock info.